pychromecast/__init__.py

```diff
@@ -161,31 +178,72 @@ class Chromecast(object):
     """
     Class to interface with a ChromeCast.
 
+    :param port: The port to use when connecting to the device, set to None to
+                 use the default of 8009. Special devices such as Cast Groups
+                 may return a different port number so we need to use that.
+    :param device: DeviceStatus with initial information for the device.
+    :type device: pychromecast.dial.DeviceStatus
     :param tries: Number of retries to perform if the connection fails.
                   None for inifinite retries.
+    :param timeout: A floating point number specifying the socket timeout in
+                    seconds. None means to use the default which is 30 seconds.
     :param retry_wait: A floating point number specifying how many seconds to
                        wait between each retry. None means to use the default
                        which is 5 seconds.
     """
 
-    def __init__(self, host, device=None, tries=None, retry_wait=None):
+    def __init__(self, host, port=None, device=None, **kwargs):
+        tries = kwargs.pop('tries', None)
+        timeout = kwargs.pop('timeout', None)
+        retry_wait = kwargs.pop('retry_wait', None)
+
         self.logger = logging.getLogger(__name__)
 
         # Resolve host to IP address
         self.host = host
+        self.port = port or 8009
 
         self.logger.info("Querying device status")
-        self.device = device or get_device_status(self.host)
+        self.device = device
+        if device:
+            dev_status = get_device_status(self.host)
+            if dev_status:
+                self.device = DeviceStatus(
+                    friendly_name=(device.friendly_name or
+                                   dev_status.friendly_name),
+                    model_name=(device.model_name or
+                                dev_status.model_name),
+                    manufacturer=(device.manufacturer or
+                                  dev_status.manufacturer),
+                    api_version=(device.api_version or
+                                 dev_status.api_version),
+                    uuid=(device.uuid or
+                          dev_status.uuid),
+                    cast_type=(device.cast_type or
+                               dev_status.cast_type),
+                )
+            else:
+                self.device = DeviceStatus(
```

Comment:
Ah, I did that to get a copy but since these are tuples there's no point in that. I'll fix.